hearth/lib/hearth/interceptor_list.rb

```diff
@@ -35,6 +35,8 @@ def add(interceptor)
 
     alias << add
 
+    # TODO: remove this from public api
```

Comment:
Why? I think it must be part of the public API if its part of Config? (unless we move away from using the List classes and uses some of List of Type validation?)

Reply:
Open to ideas but I don't think this should be publicly callable in a public API (List)

Reply:
Ah yeah - I see what you mean.  The functionality could be moved into a utilility in... middleware maybe?

I'm also still a bit on the fence about having a class for this list in the ifrst place.  It seems like we should just have a List and add better validation utilities (which we now have the ability to do in codegen).

Reply:
I think the InterceptorList makes sense. We could possible do this code in the Interceptor module (api private). I will investigate in a follow up PR - I have a list of small follow ups that I found during this.